src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -862,12 +874,16 @@ public Boolean call() throws InterruptedException {
                 System.out.println(Messages.SSHLauncher_LaunchCompletedDuration(getTimestamp(),
                         nodeName, host, duration));
             }
-            executorService.shutdown();
         } catch (InterruptedException e) {
             System.out.println(Messages.SSHLauncher_LaunchFailed(getTimestamp(),
                     nodeName, host));
+        } finally {
+            ExecutorService srv = launcherExecutorService;
+            if (srv != null) {
+                srv.shutdown();
+                launcherExecutorService = null;
```

Comment:
Maybe swap these two lines to avoid a race condition?

Reply:
makes sense assuming that `srv.shutdown()` never throws RuntimeException. It should not according to https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html#shutdown(). Maybe it also makes sense to use https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html#shutdownNow(), we can abort this executor service right away